### ISSUE TITLE
Call to `NewAddress` while filters are fetchin causes deadlock

### ISSUE THREAD

**Issue by halseth** [lightninglabs/neutrino#91]
Currently in the Lightning App, we'll query for a new address at startup, both on initial startup and on catchup after the app has been offline. This causes a deadlock

```
1 @ 0x102ea0a 0x102eabe 0x10059db 0x1005775 0x14c011e 0x14d7aff 0x150425d 0x150f20e 0x1345d60 0x1503e64 0x176b527 0x18c475a 0x16fa339 0x18094e6 0x16e6657 0x16a76c8 0x16aa908 0x16b017f 0x105b421
#	0x14c011d	github.com/lightningnetwork/lnd/vendor/github.com/lightninglabs/neutrino.(*Rescan).Update+0xad				/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/lightninglabs/neutrino/rescan.go:1102
#	0x14d7afe	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/chain.(*NeutrinoClient).NotifyReceived+0x48e	/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/chain/neutrino.go:428
#	0x150425c	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet.(*Wallet).newAddress+0x3bc			/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet/wallet.go:2801
#	0x150f20d	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet.(*Wallet).NewAddress.func1+0xbd		/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet/wallet.go:2772
#	0x1345d5f	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/walletdb.Update+0x1cf				/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/walletdb/interface.go:229
#	0x1503e63	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet.(*Wallet).NewAddress+0x93			/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet/wallet.go:2769
#	0x176b526	github.com/lightningnetwork/lnd/lnwallet/btcwallet.(*BtcWallet).NewAddress+0xb6						/Users/johan/golang/src/github.com/lightningnetwork/lnd/lnwallet/btcwallet/btcwallet.go:255
#	0x18c4759	main.(*rpcServer).NewAddress+0x69											/Users/johan/golang/src/github.com/lightningnetwork/lnd/rpcserver.go:528
#	0x16fa338	github.com/lightningnetwork/lnd/lnrpc._Lightning_NewAddress_Handler.func1+0x88						/Users/johan/golang/src/github.com/lightningnetwork/lnd/lnrpc/rpc.pb.go:6287
#	0x18094e5	github.com/lightningnetwork/lnd/macaroons.(*Service).UnaryServerInterceptor.func1+0x155					/Users/johan/golang/src/github.com/lightningnetwork/lnd/macaroons/service.go:126
#	0x16e6656	github.com/lightningnetwork/lnd/lnrpc._Lightning_NewAddress_Handler+0x166						/Users/johan/golang/src/github.com/lightningnetwork/lnd/lnrpc/rpc.pb.go:6289
#	0x16a76c7	github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).processUnaryRPC+0x9c7				/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:781
#	0x16aa907	github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).handleStream+0x1317				/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:981
#	0x16b017e	github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1+0x9e			/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:551

1 @ 0x102ea0a 0x102eabe 0x103f2b4 0x103efcd 0x1063048 0x124775b 0x1247448 0x1768798 0x1768c70 0x1345d23 0x1506975 0x14ee765 0x105b421
#	0x103efcc	sync.runtime_SemacquireMutex+0x3c												/usr/local/Cellar/go/1.10.3/libexec/src/runtime/sema.go:71
#	0x1063047	sync.(*Mutex).Lock+0x107													/usr/local/Cellar/go/1.10.3/libexec/src/sync/mutex.go:134
#	0x124775a	github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*DB).beginRWTx+0x5a						/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/db.go:567
#	0x1247447	github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt.(*DB).Begin+0x37							/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/coreos/bbolt/db.go:516
#	0x1768797	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/walletdb/bdb.(*db).beginTx+0x37				/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/walletdb/bdb/db.go:281
#	0x1768c6f	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/walletdb/bdb.(*db).BeginReadWriteTx+0x2f			/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/walletdb/bdb/db.go:293
#	0x1345d22	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/walletdb.Update+0x192					/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/walletdb/interface.go:225
#	0x1506974	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet.(*Wallet).handleChainNotifications.func2+0x224	/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet/chainntfns.go:56
#	0x14ee764	github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet.(*Wallet).handleChainNotifications+0xcf4		/Users/johan/golang/src/github.com/lightningnetwork/lnd/vendor/github.com/btcsuite/btcwallet/wallet/chainntfns.go:144
```

I've been able to reliably reproduce by syncing the app, exiting, then restarting it after a few block has been mined on testnet.

The reason this is happening is that a call to `NewAddress` will open a database `RW` transaction: https://github.com/btcsuite/btcwallet/blob/7b84dc25a61634c450d21ec7f47d5e916eb88fdb/wallet/wallet.go#L2768

At the same time, the ongoing rescan needs to obtain database RW access to update the current best height: https://github.com/btcsuite/btcwallet/blob/7b84dc2/wallet/chainntfns.go#L52

The call to `NewAddress` will call the chain client's `NotifyReceived` with the new address: https://github.com/btcsuite/btcwallet/blob/7b84dc25a61634c450d21ec7f47d5e916eb88fdb/wallet/wallet.go#L2800

Which will attempt to notify the ongoing rescan about the update: https://github.com/btcsuite/btcwallet/blob/7b84dc2/chain/neutrino.go#L428

This call will block trying the send on the `updateChan`: https://github.com/lightninglabs/neutrino/blob/31332f9c7a241da72835ad33ff340c3998e30de8/rescan.go#L1102

**Comment by halseth** [comment]
This is happening in the app since we are killing an ongoing rescan to start a new one: https://github.com/btcsuite/btcwallet/blob/7b84dc2/chain/neutrino.go#L340

This causes the update that was sent to the ongoing one (and that is not yet handled) will never be received from the `update` channel, and gets stuck. 

I guess this could also mean we might lose valuable notifications in case a rescan is quit, since we exit the unfinished rescan with a non-error: https://github.com/lightninglabs/neutrino/blob/31332f9c7a241da72835ad33ff340c3998e30de8/rescan.go#L289

**Comment by Roasbeef** [comment]
Fixed by #92.
